controllers/kubevirtmachine_controller.go

```diff
@@ -275,6 +276,7 @@ func (r *KubevirtMachineReconciler) reconcileNormal(ctx *context.MachineContext)
 		// Mark VMProvisionedCondition to indicate that the VM has successfully started
 		conditions.MarkTrue(ctx.KubevirtMachine, infrav1.VMProvisionedCondition)
 	} else {
+		conditions.MarkFalse(ctx.KubevirtMachine, infrav1.VMProvisionedCondition, "notScheduled", clusterv1.ConditionSeverityInfo, "unknown reason")
```

Comment:
I think what we need here is a series of checks into _why_ the VM isn't ready.

So, think about a function here to set the Ready condition to false. something like the logic below. The idea here is we start by looking for detailed reasons the VM might not be ready, and then slowly work down to more generic reasons the VM isn't ready.

```
func mapVMStatusToMachineCondition(...) {

    if <vm has condition  reason: Unschedulable status: "False" type: PodScheduled> {
        conditions.MarkFalse(ctx.KubevirtMachine, infrav1.VMProvisionedCondition, "notScheduled", clusterv1.ConditionSeverityInfo, <reason given from vm condition> )
    } else {
      conditions.MarkFalse(ctx.KubevirtMachine, infrav1.VMProvisionedCondition, "vmNotReady", clusterv1.ConditionSeverityInfo, <reason vm condition is set to not ready> )
    }
}
```

In the future that function will keep getting expanded. The logic will look something like this from a high level

```
if vm is not schedulable {
} else if dv storage profile not found {

} else if dv pending storage provisioning {

} else if dv failure to import image { 

} else {
< generic VMProvisioned = false reason >
}
```